src/Tickets/Seating/Commerce/Controller.php

```diff
@@ -48,6 +56,68 @@ public function unregister(): void {
 			'tec_tickets_seating_timer_token_object_id_entries',
 			[ $this, 'filter_timer_token_object_id_entries' ],
 		);
+
+		remove_filter( 'tec_tickets_commerce_increase_ticket_stock', [ $this, 'sync_seated_tickets_stock' ] );
+		remove_filter( 'tec_tickets_commerce_decrease_ticket_stock', [ $this, 'sync_seated_tickets_stock' ] );
+	}
+
+	/**
+	 * Filters the stock update value for a ticket.
+	 *
+	 * @since TBD
+	 *
+	 * @param int           $stock  The updated stock value.
+	 * @param Ticket_Object $ticket The ticket object.
+	 * @param WP_Post       $order  The order post object.
+	 *
+	 * @return int The updated stock value.
+	 */
+	public function sync_seated_tickets_stock( int $stock, Ticket_Object $ticket, WP_Post $order ): int {
+		$events = $order->events_in_order ?? [];
+
+		if ( empty( $events ) ) {
+			return $stock;
+		}
+
+		$event = get_post( array_values( $events )['0'] ); // We only support one event per order for now.
+
+		if ( ! $event instanceof WP_Post || ! $event->ID ) {
+			return $stock;
+		}
+
+		$ticket_seat_key = get_post_meta( $ticket->ID, Meta::META_KEY_SEAT_TYPE, true );
+
+		// Not a seating ticket. We should not modify the stock.
+		if ( ! $ticket_seat_key ) {
+			return $stock;
+		}
+
+		$provider = Tickets::get_event_ticket_provider_object( $event->ID );
+
+		if ( ! $provider ) {
+			return $stock;
+		}
+
+		foreach ( tribe_tickets()->where( 'event', $event->ID )->get_ids( true ) as $ticket_id ) {
```

Comment:
```suggestion
		foreach ( tribe_tickets()->where( 'event', $event->ID )->get_ids( true ) as $ticket_id ) {
```

Let's use `tec_tc_tickets` instead as this only impacts TicketsCommerce..

Reply:
We may also use `meta_not_equals` clause to filter the same seated key tickets.

Reply:
Will modify to impact other providers as well, so no longer revelant :D